mitodl/micromasters
# Added reset status to financial aid

#### What are the relevant tickets?
closes #1201

#### What's this PR do?
Adds a `reset` state to the Financial Aid applications so that an user can have another application if all the others are in `reset` state

#### Where should the reviewer start?
probably from `financialaid/models.py`

#### How should this be manually tested?
Everything should work as before, but from the ADMIN you can change the financial aid to `reset` and in the dashboard the user will be able to apply again.

#### Any background context you want to provide?
This PR needs extensive manual testing especially. 
@pdpinch I think you should take care of validate that all the previous functionality and constraints are intact.
